gui/wxpython/main_window/frame.py

```diff
@@ -270,6 +270,30 @@ def IsPaneShown(self, name):
             return self._auimgr.GetPane(name).IsShown()
         return False
 
+    def _createStatusbar(self):
+        """Create main window statusbar"""
+        self.statusbar = wx.StatusBar(self, id=wx.ID_ANY)
+        self.statusbar.SetMinHeight(24)
+        self.statusbar.SetFieldsCount(2)
+        self.statusbar.SetStatusWidths([-1, 100])
+        self.mask = SbMask(self.statusbar, self._giface)
+        self._repositionStatusbar()
+
+    def _repositionStatusbar(self):
+        """Reposition widgets in main window statusbar"""
+        rect1 = self.statusbar.GetFieldRect(1)
+        rect1.x += 1
+        rect1.y += 1
+        self.mask.GetWidget().SetRect(rect1)
+
+    def OnSize(self, event):
+        """Adjust main window statusbar on changing size"""
+        self._repositionStatusbar()
+
+    def SetStatusText(self, *args):
+        """Overide wx.StatusBar method"""
+        self.statusbar.SetStatusText(*args)
+
```

Comment:
I would move this into lmgr/statusbar.py, similarly to the SbManager, frame.py is already big.

Reply:
Good point. :-) I made the class SbMain which encapsulates the creation as well as statusbar reposition..